channeldb/channel.go

```diff
@@ -3743,6 +3747,63 @@ func MakeKeyLocRecord(typ tlv.Type, keyLoc *keychain.KeyLocator) tlv.Record {
 	return tlv.MakeStaticRecord(typ, keyLoc, 8, EKeyLocator, DKeyLocator)
 }
 
+// HTLCEntry specifies the minimal info needed to be stored on disk for ALL the
+// historical HTLCs, which is useful for constructing RevocationLog when a
+// breach is detected.
+type HTLCEntry struct {
+	// RHash is the payment hash of the HTLC.
+	RHash [32]byte
+
+	// Amt is the amount of milli-satoshis this HTLC escrows.
+	Amt lnwire.MilliSatoshi
+
+	// RefundTimeout is the absolute timeout on the HTLC that the sender
+	// must wait before reclaiming the funds in limbo.
+	RefundTimeout uint32
+
+	// OutputIndex is the output index for this particular HTLC output
+	// within the commitment transaction.
+	OutputIndex int32
```

Comment:
If we wanted to save an extra 2 bytes here (int32 -> uint16), then we could store this as _just_ a 2 byte value. This gives us a max number of HTLCs of 65K (from the PoV of the commitment transaction), but we already have that limit on disk given we use a 2 byte value to signal the number of HTLCs to follow. The current hard coded protocol limit is ~800, and even if we lift that to the widely deployed _policy_ limit, you end up with 8k or so total HTLCs (need to revisit this number).